### PR TITLE
Added UT for base controller.

### PR DESCRIPTION
**Pull request by yashgit891**
### - Link added should be a valid URL and accessible throughout the org.
   ### - If the branch name contains hotfix / revert by default the BVT workflow check will pass.

| Test Case Document URL                        |
|-----------------------------------------------|
| Please paste test case document link here.... |

### CODE REVIEW

File: Test/Unit/BaseControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        
+        $this->context->shouldReceive('getResponse')->andReturn($this->response);
+
+        $this->response->shouldReceive('setStatusHeader');
+
+        $this->checkoutSession->shouldReceive('getQuote')->andReturn($this->quoteModel);
+
+        $this->checkoutFactory->shouldReceive('create')->andReturn($this->checkoutModel);
+
+        $this->baseController = \Mockery::mock(Razorpay\Magento\Controller\BaseController::class, [ $this->context,
```
Comment (garimaprachi) [inline Test/Unit/BaseControllerTest.php]: Remove space after [

File: Test/Unit/BaseControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                                                                                                    $this->customerSession,
+                                                                                                    $this->checkoutSession,
+                                                                                                    $this->config])->makePartial()->shouldAllowMockingProtectedMethods();
+        
+    }
```
Comment (garimaprachi) [inline Test/Unit/BaseControllerTest.php:59]: Add line after func end

File: Test/Unit/BaseControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        return $property->getValue($object);
+    }
+    public function testGetQuote()
+    {
+        $this->assertSame($this->quoteModel, $this->baseController->getQuote());
+    }
+    public function testGetCheckout()
+    {
+        $this->getProperty($this->baseController, 'checkoutFactory', $this->checkoutFactory);
+        $this->assertSame($this->checkoutModel, $this->baseController->getCheckout());
```
Comment (garimaprachi) [inline Test/Unit/BaseControllerTest.php:92]: Add line before asserting

File: Test/Unit/BaseControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    public function testInitCheckout()
+    {
+        $this->quoteModel->shouldReceive('hasItems')->andReturn(true);
+        $this->quoteModel->shouldReceive('getHasError')->andReturn(true);
+
+        $this->expectException("\Magento\Framework\Exception\LocalizedException");
+        $this->expectExceptionMessage("We can't initialize checkout.");
+
+        $this->baseController->initCheckout();
+
```
Comment (garimaprachi) [inline Test/Unit/BaseControllerTest.php]: Remove line before func end

File: Test/Unit/BaseControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        $this->quoteModel->shouldReceive('hasItems')->andReturn(true);
+        $this->quoteModel->shouldReceive('getHasError')->andReturn(true);
+
+        $this->expectException("\Magento\Framework\Exception\LocalizedException");
+        $this->expectExceptionMessage("We can't initialize checkout.");
+
+        $this->baseController->initCheckout();
+        
+    }
+}
```
Comment (garimaprachi) [inline Test/Unit/BaseControllerTest.php]: add line after eof

File: Test/Unit/BaseControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            \Razorpay\Magento\Model\CheckoutFactory::class
+        );
+        $this->checkoutModel = \Mockery::mock(
+            \Razorpay\Magento\Model\Checkout::class
+        );
+        $this->response = \Mockery::mock(
+            \Magento\Framework\App\ResponseInterface::class
+        );
+        
+
```
Comment (garimaprachi) [inline Test/Unit/BaseControllerTest.php:43]: Remove one line

**Review by garimaprachi — COMMENTED** [review]
LGTM

**Review by abdulwahidsharief — APPROVED** [review]
LGTM

**Pull request merged**